kolibri/plugins/learn/assets/src/views/ChannelCard.vue

```diff
@@ -0,0 +1,219 @@
+  import TextTruncator from 'kolibri.coreVue.components.TextTruncator';
+  import CardThumbnail from './ContentCard/CardThumbnail';
+
+  export default {
+    name: 'ChannelCard',
+    components: {
+      CardThumbnail,
+      CoachContentLabel,
+      TextTruncator,
+    },
+    props: {
+      title: {
+        type: String,
+        required: true,
+      },
+      tagline: {
+        type: String,
+        required: false,
+      },
+      thumbnail: {
+        type: String,
+        required: false,
+      },
+      kind: {
+        type: String,
+        required: true,
+        validator: validateContentNodeKind,
+      },
+      showContentIcon: {
+        type: Boolean,
+        default: true,
+      },
+      // ContentNode.coach_content will be `0` if not a coach content leaf node,
+      // or a topic without coach content. It will be a positive integer if a topic
+      // with coach content, and `1` if a coach content leaf node.
+      numCoachContents: {
+        type: Number,
+        default: 0,
+      },
+      progress: {
+        type: Number,
+        required: false,
+        default: 0.0,
+        validator(value) {
+          return value >= 0.0 && value <= 1.0;
+        },
+      },
+      link: {
+        type: Object,
+        required: true,
+        validator: validateLinkObject,
+      },
+      isMobile: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    computed: {
+      ...mapGetters(['isLearner', 'isUserLoggedIn']),
+      isTopic() {
+        return this.kind === ContentNodeKinds.TOPIC || this.kind === ContentNodeKinds.CHANNEL;
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="scss" scoped>
+
+  @import '~kolibri-design-system/lib/styles/definitions';
+  @import './ContentCard/card';
+
+  $margin: 16px;
+
+  .coach-content-label {
+    display: inline-block;
+  }
+
+  .card {
+    @extend %dropshadow-1dp;
+
+    display: inline-block;
+    width: 100%;
+    min-height: 222px; // Defined in Figma
+    margin-bottom: 24px;
+    text-decoration: none;
+    vertical-align: top;
+    border-radius: $radius;
+    transition: box-shadow $core-time ease;
+    &:hover {
+      @extend %dropshadow-8dp;
+    }
+    &:focus {
+      outline-width: 4px;
+      outline-offset: 6px;
+    }
+  }
+
+  .card-heading {
+    padding: 0 $margin !important;
+    border-bottom: 2px solid #cecece;
+  }
+
+  .card-content {
+    width: 100%;
+    // Height set to ensure consistent text height
+    // calculated from 150
+    height: 172px;
+    padding: $margin;
+  }
+
+  .thumbnail {
+    position: relative;
+    display: inline-block;
+  }
+
+  .text {
+    position: relative;
+    display: inline-block;
+    padding: 0 0 0 $margin;
+    margin: 0;
+    vertical-align: top;
+  }
+
+  .card-footer {
+    display: block;
+    width: 100%;
+    height: 48px;
+    padding: $margin;
+    font-size: 12px;
+  }
+
+  /deep/.card-thumbnail-wrapper {
+    max-width: 100%;
+  }
```

Comment:
this looks potentially brittle 🤔

Reply:
The issue is that CardThumbnail sets hard widths and heights so text starts to overlap the image when its KFixedGridItem wrapper shrinks smaller than the fixed height/width img.

I only tested this working properly in Chrome but I'd love some additional testing and eyes on the responsiveness in particular.